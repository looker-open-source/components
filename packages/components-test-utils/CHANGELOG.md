# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.10.4](https://github.com/looker-open-source/components/compare/v0.10.3...v0.10.4) (2021-01-25)

**Note:** Version bump only for package @looker/components-test-utils





## [0.10.3](https://github.com/looker-open-source/components/compare/v0.10.2...v0.10.3) (2021-01-25)

**Note:** Version bump only for package @looker/components-test-utils





## [0.10.2](https://github.com/looker-open-source/components/compare/v0.10.1...v0.10.2) (2021-01-22)


### Bug Fixes

* On build properly set package.json "files" key ([#1806](https://github.com/looker-open-source/components/issues/1806)) ([c81c4bb](https://github.com/looker-open-source/components/commit/c81c4bb625c58ede49957aad90ac8d9d7b2c4b79))





## [0.10.1](https://github.com/looker-open-source/components/compare/v0.10.0...v0.10.1) (2021-01-22)


### Bug Fixes

* Properly build ems/cjs/ts for easy deep-imports ([#1804](https://github.com/looker-open-source/components/issues/1804)) ([2a523cd](https://github.com/looker-open-source/components/commit/2a523cd70b079944376c6dc87c18202e05a97b01))





# [0.10.0](https://github.com/looker-open-source/components/compare/v0.9.29...v0.10.0) (2021-01-19)


### Bug Fixes

* Properly configure build for tree-shaking support ([9202b72](https://github.com/looker-open-source/components/commit/9202b72bc5fb99eb3e1af7d9e56f3dc15b1df2a4))





## [0.9.15]

### Changed

- `@looker/components-test-utils` helpers will no longer produce `globalStyles`
