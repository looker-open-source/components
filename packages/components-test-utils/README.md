# Test Utilities

Shared functions used for mocking and testing Looker UI Components.
