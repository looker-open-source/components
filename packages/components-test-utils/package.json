{
  "name": "@looker/components-test-utils",
  "license": "MIT",
  "version": "0.9.12",
  "main": "lib/index.js",
  "types": "lib/index.d.ts",
  "files": [
    "lib"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/looker-open-source/components",
    "directory": "packages/components-test-utils"
  },
  "publishConfig": {
    "access": "public"
  },
  "devDependencies": {
    "@looker/components-providers": "^0.9.12",
    "@testing-library/react": "^10.4.8",
    "@types/enzyme": "^3.10.5",
    "@types/jest": "^26.0.10",
    "@types/react": "^16.9.49",
    "@types/react-test-renderer": "^16.9.3",
    "enzyme": "^3.11.0",
    "jest-styled-components": "^6.3.4",
    "react": "^16.13.1",
    "react-test-renderer": "^16.13.1",
    "styled-components": "^4.4.1"
  },
  "peerDependencies": {
    "react": "^16.11"
  },
  "gitHead": "62febc162e3e45124f403e2c84fc741f68fe6714"
}
