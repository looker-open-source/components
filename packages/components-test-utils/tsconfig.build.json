{
  "extends": "../../tsconfig.build.json",
  "compilerOptions": {
    "rootDir": "./src",
    "outDir": "./lib"
  },
  "include": ["../../types", "src/**/*"]
}
