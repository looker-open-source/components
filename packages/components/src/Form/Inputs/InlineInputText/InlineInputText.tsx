/*

 MIT License

 Copyright (c) 2020 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */

import pick from 'lodash/pick'
import React, { ChangeEvent, forwardRef, Ref, useState } from 'react'
import isFunction from 'lodash/isFunction'
import styled from 'styled-components'
import { typography, TypographyProps } from '@looker/design-tokens'
import { inputPropKeys, InputProps } from '../InputProps'

interface InlineInputTextProps
  extends TypographyProps,
    Omit<InputProps, 'type'> {
  underlineOnlyOnHover?: boolean
  value?: string
  simple?: boolean
}

export const InlineInputTextInternal = forwardRef(
  (
    {
      className,
      onChange,
      underlineOnlyOnHover,
      value: valueProp,
      placeholder,
      simple = false,
      ...props
    }: InlineInputTextProps,
    ref: Ref<HTMLInputElement>
  ) => {
    const [value, setValueChange] = useState(valueProp || '')

    const displayValue = isFunction(onChange) ? valueProp : value

    const handleValueChange = (event: ChangeEvent<HTMLInputElement>) => {
      setValueChange(event.currentTarget.value)
    }

    const handleChange = isFunction(onChange) ? onChange : handleValueChange

    /**
     * &#8203; is a zero-width space â€“ ensures that visibleText gets accurate line-height applied
     */

    return (
      <div className={className} data-testid="inlineInputText">
        <Input
          onChange={handleChange}
          simple={simple}
          underlineOnlyOnHover={underlineOnlyOnHover}
          value={displayValue}
          ref={ref}
          {...pick(props, inputPropKeys)}
        />
        <VisibleText displayValue={displayValue}>
          {displayValue || placeholder}&#8203;
        </VisibleText>
      </div>
    )
  }
)

InlineInputTextInternal.displayName = 'InlineInputTextInternal'

const Input = styled.input.attrs({ type: 'text' })<InlineInputTextProps>`
  background: transparent;
  border: none;
  color: transparent;
  font: inherit;
  caret-color: ${(props) => props.theme.colors.palette.charcoal900};
  height: 100%;
  left: 0;
  outline: none;
  padding: 0;
  position: absolute;
  text-transform: inherit;
  top: 0;
  width: 100%;
`

interface VisibleTextProps {
  displayValue?: string
}
const VisibleText = styled.div<VisibleTextProps>`
  color: ${({ displayValue, theme }) =>
    displayValue
      ? theme.colors.palette.charcoal900
      : theme.colors.palette.charcoal400};
`

export const InlineInputText = styled(InlineInputTextInternal)`
  ${typography}

  border: none;
  border-bottom: 1px dashed;
  border-bottom-color: ${(props) =>
    props.underlineOnlyOnHover || props.simple
      ? 'transparent'
      : props.theme.colors.palette.charcoal300};
  display: inline-flex;
  flex-direction: column;
  justify-content: center;
  position: relative;
  min-width: 2rem;

  :focus,
  :hover {
    outline: none;
    border-bottom-color: ${(props) => props.theme.colors.palette.purple400};
    background-color: ${(props) => props.theme.colors.palette.charcoal100};
  }

  :focus {
    border-bottom-style: solid;
  }
`
