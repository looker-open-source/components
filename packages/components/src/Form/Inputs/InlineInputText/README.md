# InlineInputText

An input styled as text, with an underline and hover styles to indicate it is editable.

`InlineInputText` implements `InputText` internally â€“ see [that component's documentation](?path=/docs/docs-inputtext--docs) for more information.

```tsx
function HeadingParagraph() {
  return (
    <Space around>
      <Heading>
        <InlineInputText value="Type here..." />
      </Heading>
      <Paragraph color="text1">
        <InlineInputText placeholder="Type here..." />
      </Paragraph>
    </Space>
  );
}
```

## Simple

Use the `simple` prop to remove the underline except on hover and focus.

```tsx
function Simple() {
  return <InlineInputText simple value="Some text here" />;
}
```

## Disabled

Use the `disabled` prop to make the input uneditable and greyed out.

<Editor code={storyDisabled} />

```tsx
function Disabled() {
  return <InlineInputText disabled value="You can't change me." />;
}
```

## ReadOnly

Use the `readOnly` prop to make the input uneditable and have no underline.

<Editor code={storyReadOnly} />

```tsx
function ReadOnly() {
  return <InlineInputText readOnly value="You can't change me." />;
}
```
