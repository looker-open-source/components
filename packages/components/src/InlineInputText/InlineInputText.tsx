/*
 MIT License
 Copyright (c) 2019 Looker Data Sciences, Inc.
 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:
 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.
 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.
 */

import pick from 'lodash/omit'
import React, { forwardRef, Ref } from 'react'
import isFunction from 'lodash/isFunction'
import styled from 'styled-components'
import { typography, TypographyProps } from '@looker/design-tokens'
import { inputPropKeys, InputProps } from '../Form/Inputs/InputProps'

interface InlineInputTextProps
  extends TypographyProps,
    Omit<InputProps, 'type'> {
  underlineOnlyOnHover?: boolean
  value?: string
}

export const InlineInputTextInternal = forwardRef(
  (
    {
      className,
      onChange,
      underlineOnlyOnHover,
      value,
      ...props
    }: InlineInputTextProps,
    ref: Ref<HTMLInputElement>
  ) => {
    const [valueChange, setValueChange] = React.useState(value || '')

    const displayValue = isFunction(onChange) ? value : valueChange

    const handleValueChange = (event: React.ChangeEvent<HTMLInputElement>) => {
      setValueChange(event.currentTarget.value)
    }

    const handleChange = isFunction(onChange) ? onChange : handleValueChange

    return (
      <span className={className}>
        <Input
          onChange={handleChange}
          underlineOnlyOnHover={underlineOnlyOnHover}
          value={displayValue}
          ref={ref}
          size={1}
          {...pick(props, inputPropKeys)}
        />
        <HiddenText>{displayValue}</HiddenText>
      </span>
    )
  }
)

InlineInputTextInternal.displayName = 'InlineInputTextInternal'

const Input = styled.input.attrs({ type: 'text' })<InlineInputTextProps>`
  background: transparent;
  border: none;
  border-bottom: 1px dashed;
  border-bottom-color: ${props =>
    props.underlineOnlyOnHover
      ? 'transparent'
      : props.theme.colors.palette.charcoal300};
  padding: 0;
  font: inherit;
  color: inherit;
  text-transform: inherit;
  width: 100%;

  :focus,
  :hover {
    outline: none;
    border-bottom-color: ${props => props.theme.colors.palette.purple400};
    background-color: ${props => props.theme.colors.palette.charcoal100};
  }

  :focus {
    border-bottom-style: solid;
  }
`

const HiddenText = styled.span`
  height: 0;
  overflow: hidden;
  white-space: pre-wrap;
  padding: 0 1px;
`

export const InlineInputText = styled(InlineInputTextInternal)`
  ${typography}

  display: inline-flex;
  flex-direction: column;
  justify-content: center;
`
