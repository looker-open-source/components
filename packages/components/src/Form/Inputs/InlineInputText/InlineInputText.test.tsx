/*

 MIT License

 Copyright (c) 2020 Looker Data Sciences, Inc.

 Permission is hereby granted, free of charge, to any person obtaining a copy
 of this software and associated documentation files (the "Software"), to deal
 in the Software without restriction, including without limitation the rights
 to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 copies of the Software, and to permit persons to whom the Software is
 furnished to do so, subject to the following conditions:

 The above copyright notice and this permission notice shall be included in all
 copies or substantial portions of the Software.

 THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.

 */

import { assertSnapshot, renderWithTheme } from '@looker/components-test-utils'
import { screen } from '@testing-library/react'
import React from 'react'
import { InlineInputText, InlineInputTextBase } from './InlineInputText'

test('InlineInputText renders an input with a placeholder', () => {
  assertSnapshot(<InlineInputText placeholder="this is the placeholder" />)
})

test('InlineInputText renders an input with no value', () => {
  assertSnapshot(<InlineInputText />)
})

test('InlineInputText renders an input with specific predefined value', () => {
  assertSnapshot(<InlineInputText value="type here..." />)
})

test('InlineInputTextBase renders a subset of styles', () => {
  renderWithTheme(<InlineInputTextBase placeholder="type here..." />)
  expect(
    screen.getByPlaceholderText('type here...').parentElement
  ).not.toHaveStyle('border-bottom: 1px dashed')
})
