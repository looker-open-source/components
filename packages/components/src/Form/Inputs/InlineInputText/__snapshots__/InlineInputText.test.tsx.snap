// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineInputText renders an input with a placeholder 1`] = `
.c1 {
  background: transparent;
  border: none;
  caret-color: #262D33;
  color: inherit;
  height: 100%;
  outline: none;
  width: 100%;
  font: inherit;
  left: 0;
  padding: 0;
  position: absolute;
  text-align: inherit;
  text-transform: inherit;
  top: 0;
}

.c1::-webkit-search-decoration,
.c1::-webkit-search-cancel-button,
.c1::-webkit-search-results-button,
.c1::-webkit-search-results-decoration {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c1::-webkit-input-placeholder {
  color: #939BA5;
}

.c1::-moz-placeholder {
  color: #939BA5;
}

.c1:-ms-input-placeholder {
  color: #939BA5;
}

.c1::placeholder {
  color: #939BA5;
}

.c1::-webkit-outer-spin-button,
.c1::-webkit-inner-spin-button {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c1[type='number'] {
  -webkit-appearance: textfield;
  -moz-appearance: textfield;
  appearance: textfield;
}

.c2 {
  color: transparent;
  line-height: inherit;
  text-align: inherit;
  white-space: pre;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-width: 2rem;
  position: relative;
  border: none;
  border-bottom: 1px dashed;
  border-bottom-color: #C1C6CC;
  color: inherit;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: inherit;
}

.c0:focus,
.c0:hover {
  background-color: #F5F6F7;
  border-bottom-color: #6C43E0;
  outline: none;
}

.c0:focus {
  border-bottom-style: solid;
}

<span
  className="c0"
>
  <input
    className="c1"
    onChange={[Function]}
    placeholder="this is the placeholder"
    type="text"
    value=""
  />
  <span
    className="c2"
  >
    this is the placeholder
  </span>
</span>
`;

exports[`InlineInputText renders an input with no value 1`] = `
.c1 {
  background: transparent;
  border: none;
  caret-color: #262D33;
  color: inherit;
  height: 100%;
  outline: none;
  width: 100%;
  font: inherit;
  left: 0;
  padding: 0;
  position: absolute;
  text-align: inherit;
  text-transform: inherit;
  top: 0;
}

.c1::-webkit-search-decoration,
.c1::-webkit-search-cancel-button,
.c1::-webkit-search-results-button,
.c1::-webkit-search-results-decoration {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c1::-webkit-input-placeholder {
  color: #939BA5;
}

.c1::-moz-placeholder {
  color: #939BA5;
}

.c1:-ms-input-placeholder {
  color: #939BA5;
}

.c1::placeholder {
  color: #939BA5;
}

.c1::-webkit-outer-spin-button,
.c1::-webkit-inner-spin-button {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c1[type='number'] {
  -webkit-appearance: textfield;
  -moz-appearance: textfield;
  appearance: textfield;
}

.c2 {
  color: transparent;
  line-height: inherit;
  text-align: inherit;
  white-space: pre;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-width: 2rem;
  position: relative;
  border: none;
  border-bottom: 1px dashed;
  border-bottom-color: #C1C6CC;
  color: inherit;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: inherit;
}

.c0:focus,
.c0:hover {
  background-color: #F5F6F7;
  border-bottom-color: #6C43E0;
  outline: none;
}

.c0:focus {
  border-bottom-style: solid;
}

<span
  className="c0"
>
  <input
    className="c1"
    onChange={[Function]}
    type="text"
    value=""
  />
  <span
    className="c2"
  >
     
  </span>
</span>
`;

exports[`InlineInputText renders an input with specific predefined value 1`] = `
.c1 {
  background: transparent;
  border: none;
  caret-color: #262D33;
  color: inherit;
  height: 100%;
  outline: none;
  width: 100%;
  font: inherit;
  left: 0;
  padding: 0;
  position: absolute;
  text-align: inherit;
  text-transform: inherit;
  top: 0;
}

.c1::-webkit-search-decoration,
.c1::-webkit-search-cancel-button,
.c1::-webkit-search-results-button,
.c1::-webkit-search-results-decoration {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c1::-webkit-input-placeholder {
  color: #939BA5;
}

.c1::-moz-placeholder {
  color: #939BA5;
}

.c1:-ms-input-placeholder {
  color: #939BA5;
}

.c1::placeholder {
  color: #939BA5;
}

.c1::-webkit-outer-spin-button,
.c1::-webkit-inner-spin-button {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c1[type='number'] {
  -webkit-appearance: textfield;
  -moz-appearance: textfield;
  appearance: textfield;
}

.c2 {
  color: transparent;
  line-height: inherit;
  text-align: inherit;
  white-space: pre;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-width: 2rem;
  position: relative;
  border: none;
  border-bottom: 1px dashed;
  border-bottom-color: #C1C6CC;
  color: inherit;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: inherit;
}

.c0:focus,
.c0:hover {
  background-color: #F5F6F7;
  border-bottom-color: #6C43E0;
  outline: none;
}

.c0:focus {
  border-bottom-style: solid;
}

<span
  className="c0"
>
  <input
    className="c1"
    onChange={[Function]}
    type="text"
    value="type here..."
  />
  <span
    className="c2"
  >
    type here...
  </span>
</span>
`;
